classes/UpgradeSelfCheck.php

```diff
@@ -465,4 +480,61 @@ public function runPrestaShopCoreChecks()
 
         return true;
     }
+
+    /**
+     * Get php min and max compatible versions for target prestashop version
+     *
+     * @return array
+     */
+    public function phpCompatibleVersions()
+    {
+        $previousShopVersion = false;
+        $length = count(self::PHP_PS_VERSIONS);
+        $i = 0;
+        $currentShopVersion = null;
+
+        foreach (self::PHP_PS_VERSIONS as $prestashopVersion => $phpVersions) {
+            if (!$previousShopVersion) {
+                $previousShopVersion = $prestashopVersion;
+                $i++;
+                continue;
+            }
+
+            // we cannot use substr to compare versions, because we can't guess the length of the current version,
+            // nor the length of the version in self::PHP_PS_VERSIONS
+            // so we loop through versions in self::PHP_PS_VERSIONS and check that our shop version is greater
+            // or equal to the version from previous iteration, and inferior to the version from current iteration
+            // (because php versions can only change for major or mineur prestashop versions, and compare_version
+            // doesn't consider 1.7.7 and 1.7.7.1 as being equal, there's no notion of precision)
+            if (version_compare($this->upgrader->version_num, $previousShopVersion, '>=')
+                && version_compare($this->upgrader->version_num, $prestashopVersion, '<')) {
+                $currentShopVersion = $previousShopVersion;
+                break;
+            }
+
+            // special case if we are in the last iteration
+            if ($i == ($length - 1) && version_compare($this->upgrader->version_num, $prestashopVersion, '>=')) {
+                $currentShopVersion = $prestashopVersion;
+                break;
+            }
+
+            $previousShopVersion = $prestashopVersion;
+            $i++;
+        }
+
+        return !empty($currentShopVersion) ? self::PHP_PS_VERSIONS[$currentShopVersion] : [];
+    }
+
+    /**
+     * Check if the current php version is compatible with PrestaShop target version
+     *
+     * @return bool
+     */
+    public function isPhpCompatible()
+    {
+        $phpCompatibleVersions = $this->phpCompatibleVersions();
+
+        $phpVersion = substr(PHP_VERSION,0,3);
```

Comment:
Maybe I'm worrying to soon but this won't work for PHP 10 😅

Reply:
I think it follows the model, from memory a screenshot with ` 7.3.33-1+ubuntu21.04.1+deb.sury.org+1` had the model changed to 7.X 

Why not use the full version name ?